daemon/api.go

```diff
@@ -1647,7 +1647,7 @@ func splitQS(qs string) []string {
 
 func getSnapConf(c *Command, r *http.Request, user *auth.UserState) Response {
 	vars := muxVars(r)
-	snapName := snap.DropNick(vars["name"])
+	snapName := ifacestate.RemapSnapFromRequest(vars["name"])
```

Comment:
For configuration we *may* need our own set of of remap helpers. Last time we discussed this the idea was that we map all configuration from "system" -> "core" even on a core18 system. If this has changed thats fine but we probably still need to tweak things a bit because e.g. "canConfigure()" allows setting "core" even if core is not installed. If we change where the "system" configuration is stored we probably need to update "canConfigure"

Reply:
it feels also strange to use a helper from ifacestate in configuration related code

Reply:
One simple fix would be to have similar remap helpers in configstate.

Reply:
This is incorrect application of IMO correct logic. This simple tweak ought to fix this:
```
snapName := ifacestate.RemapSnapToStore(ifacestate.RemapSnapFromRequest(vars["name"]))
```